Tests/ConfidenceTests/ConfidenceTest.swift

```diff
@@ -581,6 +581,45 @@ class ConfidenceTest: XCTestCase {
             XCTAssertEqual(error as? ConfidenceError, ConfidenceError.invalidContextInMessage)
         }
     }
+
+    func testRequestTimedOut() async throws {
+        class FakeClient: ConfidenceResolveClient {
+            var resolvedValues: [ResolvedValue] = []
+            func resolve(ctx: ConfidenceStruct) async throws -> ResolvesResult {
+                try await Task.sleep(nanoseconds: 5000000000)
+                return .init(resolvedValues: resolvedValues, resolveToken: "token")
+            }
+        }
+
+        let client = FakeClient()
+        client.resolvedValues = [
+            ResolvedValue(
+                variant: "default",
+                value: .init(structure: ["size": .init(integer: 3)]),
+                flag: "flag",
+                resolveReason: .match)
+        ]
+
+        let confidence = Confidence.Builder(clientSecret: "test")
+            .withContext(initialContext: ["targeting_key": .init(string: "user2")])
+            .withFlagResolverClient(flagResolver: client)
+            .withFlagApplier(flagApplier: flagApplier)
+            .withTimeout(timeout: 1)
+            .build()
+
+        try await confidence.fetchAndActivate()
+        let evaluation = try confidence.getEvaluation(
+            key: "flag.size",
+            defaultValue: 0)
+
+        XCTAssertEqual(evaluation.value, 3)
```

Comment:
what are we testing here? first, the timeout we implemented here is on the session right? basically it's under the network client and the network request, how do we want to test it here? and also we are asserting here that the returned value is 3 which is what the client is returning so I assume we are not timing out?

Reply:
We're testing that the request times out and the value stays a default 🤔 Basically my thinking was that the value should stay the same after fetch&activate because of the timeout 🤔

Reply:
in [this line](https://github.com/spotify/confidence-sdk-swift/pull/160/files/f358cc9bcd12148a26ca26d51480c494d6c57f9b..9870a6d8901f9639cbbebf411f7dfcbf2f564611#diff-5bc3ccf5821e481bbe527e2e28de64ee90fc5e0fc2095494eb878f2b8f78e212R595) we are assigning the values for client, so the clients is returning this values. the default value is 0. plus, fetchAndActivate waits as long as the client fetch takes, which is the sleep.

Reply:
and as I mentioned in the first comment, I don't understand what is going to cancel the client request since the timeout was defined in the network client itself, client is one level above that timeout. so except you test integration test OR pass a session to the client, (and update your implementation to set the timeout even when the session is passed to the class), we can't properly test that imo.

Reply:
i'll try to do an integration test then 👍